packages/core/strapi/lib/services/__tests__/entity-validator.test.js

```diff
@@ -592,4 +589,685 @@ describe('Entity validator', () => {
+          private: true,
+        },
+        updatedBy: {
+          type: 'relation',
+          relation: 'oneToOne',
+          target: 'admin::user',
+          configurable: false,
+          writable: false,
+          visible: false,
+          useJoinTable: false,
+          private: true,
+        },
+      },
+    });
+    models.set('api::category.category', {
+      kind: 'collectionType',
+      collectionName: 'categories',
+      modelType: 'contentType',
+      modelName: 'category',
+      connection: 'default',
+      uid: 'api::category.category',
+      apiName: 'category',
+      globalId: 'Category',
+      info: {
+        displayName: 'Category',
+        singularName: 'category',
+        pluralName: 'categories',
+        description: '',
+        name: 'Category',
+      },
+      attributes: {
+        name: {
+          type: 'string',
+          pluginOptions: {
+            i18n: {
+              localized: true,
+            },
+          },
+        },
+      },
+    });
+    models.set('basic.dev-compo', {
+      collectionName: 'components_basic_dev_compos',
+      uid: 'basic.dev-compo',
+      category: 'basic',
+      modelType: 'component',
+      modelName: 'dev-compo',
+      globalId: 'ComponentBasicDevCompo',
+      info: {
+        displayName: 'DevCompo',
+        icon: 'allergies',
+      },
+      attributes: {
+        categories: {
+          type: 'relation',
+          relation: 'oneToMany',
+          target: 'api::category.category',
+        },
+      },
+    });
+    models.set('plugin::upload.file', {
+      collectionName: 'files',
+      info: {
+        singularName: 'file',
+        pluralName: 'files',
+        displayName: 'File',
+        description: '',
+      },
+      attributes: {
+        name: {
+          type: 'string',
+          configurable: false,
+          required: true,
+        },
+      },
+      kind: 'collectionType',
+      modelType: 'contentType',
+      modelName: 'file',
+      connection: 'default',
+      uid: 'plugin::upload.file',
+      plugin: 'upload',
+      globalId: 'UploadFile',
+    });
+
+    const IDsThatExist = [1, 2, 3, 4, 5, 6];
+    const nonExistentIDs = [10, 11, 12, 13, 14, 15, 16];
+    const strapi = {
+      components: {
+        'basic.dev-compo': {},
+      },
+      db: {
+        query() {
+          return {
+            count: ({
+              where: {
+                id: { $in },
+              },
+            }) => IDsThatExist.filter((value) => $in.includes(value)).length,
+          };
+        },
+      },
+      errors: {
+        badRequest: jest.fn(),
+      },
+      getModel: (uid) => models.get(uid),
+    };
+
+    describe('Attribute', () => {
+      describe('Success', () => {
+        const testData = [
+          [
+            'Connect',
+            {
+              categories: {
+                disconnect: [],
+                connect: [
+                  {
+                    id: sample(IDsThatExist),
```

Comment:
When using randomization inside tests, we are exposed to the risk of having the tests pass sometimes and sometimes not. What do you think of having always the same ids instead?